crates/common/src/deposit/mod.rs

```diff
@@ -1,15 +1,19 @@
 //! Tramsaction types for Optimism.
 
 mod source;
+use alloy_consensus::Transaction;
 pub use source::{
     DepositSourceDomain, DepositSourceDomainIdentifier, L1InfoDepositSource, UpgradeDepositSource,
     UserDepositSource,
 };
 
 use alloy_primitives::B256;
 
+/// Identifier for an Optimism deposit transaction
+pub const DEPOSIT_TX_TYPE_ID: u8 = 126; // 0x7E
+
 /// A trait representing a deposit transaction with specific attributes.
-pub trait DepositTransaction {
+pub trait DepositTransaction: Transaction {
```

Comment:
fyi the is_deposit can now use a default impl

Reply:
true thanks. can remove it from this trait even.